src/middleware/company.authentication.middleware.ts

```diff
@@ -0,0 +1,38 @@
+import { NextFunction, Request, Response } from "express";
+import { authMiddleware, AuthOptions } from "@companieshouse/web-security-node";
+import { CHS_URL } from "../utils/properties";
+import { logger } from "../utils/logger";
+
+const COMPANY_NUMBER_MATCHER = new RegExp("\\/company\\/([0-9a-zA-Z]{8})");
+
+export const companyAuthenticationMiddleware = (req: Request, res: Response, next: NextFunction) => {
+
+  const companyNumber: string | undefined = extractCompanyNumberFromRequest(req);
+
+  if (!companyNumber) {
+    logger.errorRequest(req, "No Valid Company Number in URL: " + req.originalUrl);
+    return next(new Error("No Valid Company Number in Request"));
+  }
+
+  const authMiddlewareConfig: AuthOptions = {
+    chsWebUrl: CHS_URL,
+    returnUrl: req.originalUrl,
+    companyNumber: companyNumber
+  };
+
+  return authMiddleware(authMiddlewareConfig)(req, res, next);
+};
+
+const extractCompanyNumberFromRequest = (req: Request): string | undefined => {
```

Comment:
you can get the url params from req.params, but you will need to set the app.use with the param name (like in spring)
something liek this:
app.use(urls.CONFIRMATION_STATEMENT + "/company/:companyNumber", companyAuthenticationMiddleware);

have a look at https://stackabuse.com/get-query-strings-and-parameters-in-express-js/ (the extracting route params section)

i haven't actually got it running myself but should be doable

Reply:
req.params isn't populated for middleware.
This is because its the router that populates that field, and the request hasn't been through the router for middleware such as this.

Reply:
the actual route handler will be able to access that though as the url is defined with a path paramater: https://github.com/companieshouse/confirmation-statement-web/pull/65/files#diff-b1c71f3697315abae06b8eaaf77b9f90eb1c41012510bd82b92b4a0dd2632941R13

Reply:
ok